### PR TITLE
Use setup.cfg for package metadata

### PR DESCRIPTION
**Pull request by astrofrog**
As done for some of the other packages, this migrates the package metadata to ``setup.cfg``

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+[bdist_wheel]
+universal = 1
+
+[options]
+zip_safe = False
+packages = find:
+install_requires =
+	sphinx>=1.7
+	astropy-sphinx-theme
+	numpydoc
```
Comment (bsipocz) [inline setup.cfg]: don't we have a minimum version number for this, too?

File: setup.py [hunk 2]
```diff
@@ -2,29 +2,15 @@
...
+import sys
+import setuptools
+from distutils.version import LooseVersion
+from setuptools import setup
 
-with open('README.rst') as infile:
-    long_description = infile.read()
+# Setuptools 30.3.0 or later is needed for setup.cfg options to be used
+if LooseVersion(setuptools.__version__) < LooseVersion('30.3.0'):
+    sys.stderr.write("ERROR: sphinx-automodapi requires setuptools 30.3.0 or "
```
Comment (Cadair) [inline setup.py]: I feel like this error should just be "SHHHAMMMMEEE, update your stuff" at this point...

Comment (bsipocz) [reply setup.py]: Borderline, it's still not 3 years old. Once everyone is moved on from python2, maybe you can say that...

File: setup.cfg (hunk 1 above)
Comment (astrofrog) [reply setup.cfg]: Not that I know of - at some point I can try building the astropy docs with old versions to see how far back we can go before it fails but I don't have time this week.

Comment (astrofrog) [reply setup.cfg]: I opened an issue here: https://github.com/astropy/sphinx-astropy/issues/28

Comment (bsipocz) [reply setup.cfg]: thanks, that's sounds the right approach, and logically would also fit nicely with dropping python2 in the next release

File: setup.py (hunk 2 above)
Comment (Cadair) [reply setup.py]: I have the same opinion of people still using Python 2... or 3.5

Comment (bsipocz) [reply setup.py]: 🤷‍♀ we don't disagree, yet reality is another matter.

**Pull request merged**